frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/homepage/MyProjectsPortlet.java

```diff
@@ -12,42 +12,65 @@
 
 import org.eclipse.sw360.datahandler.common.CommonUtils;
 import org.eclipse.sw360.datahandler.thrift.projects.Project;
+import org.eclipse.sw360.datahandler.thrift.projects.ProjectService;
+import org.eclipse.sw360.datahandler.thrift.users.User;
 import org.eclipse.sw360.portal.portlets.Sw360Portlet;
 import org.eclipse.sw360.portal.users.LifeRayUserSession;
 import org.apache.log4j.Logger;
 import org.apache.thrift.TException;
+import org.eclipse.sw360.portal.users.UserCacheHolder;
 
 import javax.portlet.PortletException;
 import javax.portlet.RenderRequest;
 import javax.portlet.RenderResponse;
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
 
-import static org.apache.log4j.Logger.getLogger;
-
 /**
  * Small homepage portlet
  *
  * @author cedric.bodet@tngtech.com
  * @author gerrit.grenzebach@tngtech.com
+ * @author ksoranko@verifa.io
  */
 public class MyProjectsPortlet extends Sw360Portlet {
 
-    private static final Logger log = getLogger(MyProjectsPortlet.class);
+    private static final Logger LOGGER = Logger.getLogger(MyProjectsPortlet.class);
 
     @Override
     public void doView(RenderRequest request, RenderResponse response) throws IOException, PortletException {
-        List<Project> projects=null;
-
+        List<Project> myProjects = new ArrayList<>();
+        List<String> ids = new ArrayList<>();
+        User user = UserCacheHolder.getUserFromRequest(request);
+        List<Project> projects;
         try {
-            String email = LifeRayUserSession.getEmailFromRequest(request);
-            projects = thriftClients.makeProjectClient().getMyProjects(email);
+            myProjects = thriftClients.makeProjectClient().getMyProjects(user.getEmail());
         } catch (TException e) {
-            log.error("Could not fetch your projects from backend", e);
+            LOGGER.error("Could not fetch myProjects from backend for user, " + user.getEmail(), e);
         }
-
+        if(!myProjects.isEmpty()) {
+            myProjects.forEach(p -> ids.add(p.getId()));
+        }
+        try {
+            projects = thriftClients.makeProjectClient().getProjectsById(ids, user);
```

Comment:
Why do we need to load the projects again? Which information is missing? `releaseIdToUsage` ?
Maybe modify the `getMyProjects` function to receive the releases in the first step instead of calling this project endpoint twice.